utils/src/main/java/io/atomix/utils/concurrent/FailureHandlingScheduledThreadPoolExecutor.java

```
@@ -0,0 +1,63 @@
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Consumer;
+
+public class FailureHandlingScheduledThreadPoolExecutor extends ScheduledThreadPoolExecutor {
+
+  private final Consumer<Throwable> taskFailureHandler;
+
+  public FailureHandlingScheduledThreadPoolExecutor(int i, Consumer<Throwable> taskFailureHandler) {
+    super(i);
+    this.taskFailureHandler = taskFailureHandler;
+  }
+
+  public FailureHandlingScheduledThreadPoolExecutor(
+      int i, ThreadFactory threadFactory, Consumer<Throwable> taskFailureHandler) {
+    super(i, threadFactory);
+    this.taskFailureHandler = taskFailureHandler;
+  }
+
+  public FailureHandlingScheduledThreadPoolExecutor(
+      int i,
+      RejectedExecutionHandler rejectedExecutionHandler,
+      Consumer<Throwable> taskFailureHandler) {
+    super(i, rejectedExecutionHandler);
+    this.taskFailureHandler = taskFailureHandler;
+  }
+
+  public FailureHandlingScheduledThreadPoolExecutor(
+      int i,
+      ThreadFactory threadFactory,
+      RejectedExecutionHandler rejectedExecutionHandler,
+      Consumer<Throwable> taskFailureHandler) {
+    super(i, threadFactory, rejectedExecutionHandler);
+    this.taskFailureHandler = taskFailureHandler;
+  }
+
+  @Override
+  protected void afterExecute(Runnable runnable, Throwable throwable) {
+    super.afterExecute(runnable, throwable);
+    if (throwable == null && runnable instanceof Future<?>) {
+      final Future<?> future = (Future<?>) runnable;
+      if (future.isDone()) {
+        try {
+          Object result = future.get();
+        } catch (ExecutionException ee) {
+          taskFailureHandler.accept(ee.getCause());
+        } catch (InterruptedException ie) {
+          // do nothing
+        } catch (CancellationException ce) {
+          // do nothing
```

Why are these other exceptions swallowed ?